todo-app/src/App.js

```diff
@@ -1,21 +1,107 @@
-import React, { Component } from 'react';
+import React from 'react';
 import logo from './logo.svg';
 import './App.css';
+import List from './list'
 
-class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <div className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h2>Welcome to React</h2>
+  class App extends React.Component {
+    constructor (props){
+        super(props);
+        this.state = {
+          todos: [
+               {
+                description: "Read more about React",
+                done:        true
+               } ,
+               {
+              description: "Do the homework",
+               done:        false
+              },
+            ],
+          newToDo:{
+            description: "",
+            done:        false
+           }
+        }
+
+    }
+
+    handleSubmit(event){
+      event.preventDefault()
+      const {newToDo} = this.state  
+      const todos = [...this.state.todos, newToDo]
+      this.setState({todos})
+      document.getElementById("addItem").reset();
+      }
+
+    handleChange(event){
+      this.setState({
+        newToDo: {
+          description: event.target.value,
+          done: false
+        }
+      })
+    }
+
+    renderAddItem (){
+      return(
+        <div>
+          <form id="addItem" onSubmit={this.handleSubmit.bind(this)}>
+            <input type="submit" value="Add" />
+            <input type="text"  onChange={this.handleChange.bind(this)}/>
+          </form>
         </div>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
-    );
+      )
+    }
+
+    onDelete= (index) => {
+      const todosList = this.state.todos
+      todosList.splice(index, 1)
+      this.setState({todosList})
+    }
+
+
+
+    handleCheckBoxChange = (index) => {
```

Comment:
Same here in terms of writing this code with immutability in mind. You can use the slice method, and possibly [`Object.assign`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) to not mutate the todo item, which you do now by using the `index` and setting the `done` property.